component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -1658,4 +1669,140 @@ private long getUnlockTimeInMilliSeconds(AuthenticatedUser authenticatedUser) th
+        eventProperties.put(IdentityEventConstants.EventProperty.OTP_GENERATED_TIME, otpGeneratedTime);
+
+        String otpExpiryDuration = SMSOTPUtils.getTokenExpiryTime(context);
+        if (StringUtils.isNotEmpty(otpExpiryDuration)) {
+            long expiryTime = otpGeneratedTime + Long.parseLong(otpExpiryDuration);
+            eventProperties.put(IdentityEventConstants.EventProperty.OTP_EXPIRY_TIME, expiryTime);
+        }
+
+        if (StringUtils.isNotEmpty(request.getHeader(SMSOTPConstants.X_FORWARDED_FOR))) {
+            eventProperties.put(IdentityEventConstants.EventProperty.CLIENT_IP, request.getHeader(
+                    SMSOTPConstants.X_FORWARDED_FOR).split(",")[0]);
+        } else {
+            if (log.isDebugEnabled()) {
+                log.debug("Getting client-ip from remote address since x-forwarded-for is null.");
+            }
+            eventProperties.put(IdentityEventConstants.EventProperty.CLIENT_IP, request.getRemoteAddr());
+        }
+
+        Event postOtpGenEvent = new Event(IdentityEventConstants.Event.POST_GENERATE_SMS_OTP, eventProperties);
+        try {
+            SMSOTPServiceDataHolder.getInstance().getIdentityEventService().handleEvent(postOtpGenEvent);
+        } catch (IdentityEventException e) {
+            String errorMsg = "An error occurred while triggering post event in SMS OTP generation flow. " + e.getMessage();
+            throw new AuthenticationFailedException(errorMsg, e);
+        }
+    }
+
+    /**
+     * Send event after validating SMS OTP.
+     *
+     * @param request HttpServletRequest.
+     * @param context Authentication context.
+     * @throws AuthenticationFailedException Exception on authentication failure.
+     */
+    private void publishPostSMSOTPValidatedEvent(HttpServletRequest request,
+                                                 AuthenticationContext context) throws AuthenticationFailedException {
+
+        Map<String, Object> eventProperties = new HashMap<>();
+        AuthenticatedUser authenticatedUser = (AuthenticatedUser) context.getProperty(SMSOTPConstants
+                .AUTHENTICATED_USER);
+        eventProperties.put(IdentityEventConstants.EventProperty.CORRELATION_ID, context.getCallerSessionKey());
+        eventProperties.put(IdentityEventConstants.EventProperty.USER_NAME, authenticatedUser.getUserName());
+        eventProperties.put(IdentityEventConstants.EventProperty.TENANT_DOMAIN, context.getTenantDomain());
+        eventProperties.put(IdentityEventConstants.EventProperty.USER_STORE_DOMAIN, authenticatedUser
+                .getUserStoreDomain());
+        eventProperties.put(IdentityEventConstants.EventProperty.APPLICATION_NAME, context.getServiceProviderName());
+        eventProperties.put(IdentityEventConstants.EventProperty.USER_AGENT, request.getHeader(
+                SMSOTPConstants.USER_AGENT));
+
+        if (StringUtils.isNotEmpty(request.getHeader(SMSOTPConstants.X_FORWARDED_FOR))) {
```

Comment:
Can use `getClientIpAddress` in [here](https://github.com/wso2/carbon-identity-framework/blob/c4ece01ed84b5fda081bf6a5cf2715f0513e30e3/components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityUtil.java#L1056) for this purpose

Reply:
Fixed in 48dfc684db60965170330a1fab99c186531f743e